From 5963870fce25dbd70d66121b27a4647e0eee9fc8 Mon Sep 17 00:00:00 2001
From: Cyber Knight <cyberknight755@gmail.com>
Date: Tue, 16 Dec 2025 19:51:37 +0800
Subject: [PATCH 2/2] MtkGbaService: Remove remaining TelephonyMetrics
 references

- Following the previous commit, drop all references to it.

Signed-off-by: Cyber Knight <cyberknight755@gmail.com>

SX3 port: Keep const/4 v1, 0x0 before mMtkRadioExModemProxy initialization; the preceding v1 contains a SparseArray.
---
diff --git a/smali/com/mediatek/gbaservice/ril/GbaRIL.smali b/smali/com/mediatek/gbaservice/ril/GbaRIL.smali
--- a/smali/com/mediatek/gbaservice/ril/GbaRIL.smali
+++ b/smali/com/mediatek/gbaservice/ril/GbaRIL.smali
@@ -20,8 +20,6 @@
 
 .field private final mClientWakelockTracker:Lcom/android/internal/telephony/ClientWakelockTracker;
 
-.field private mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
 .field volatile mMtkRadioExModemProxy:Lvendor/mediatek/hardware/mtkradioex/modem/IMtkRadioExModem;
 
 .field final mMtkRadioExModemProxyDeathRecipient:Lcom/mediatek/gbaservice/ril/GbaRIL$MtkRadioExModemProxyDeathRecipient;
@@ -75,15 +73,6 @@
     return-void
 .end method
 
-.method static bridge synthetic -$$Nest$fgetmMetrics(Lcom/mediatek/gbaservice/ril/GbaRIL;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-    .locals 0
-
-    .line 0
-    iget-object p0, p0, Lcom/mediatek/gbaservice/ril/GbaRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    return-object p0
-.end method
-
 .method static bridge synthetic -$$Nest$mclearWakeLock(Lcom/mediatek/gbaservice/ril/GbaRIL;I)Z
     .locals 0
 
@@ -199,13 +188,6 @@
     invoke-direct {v1}, Landroid/util/SparseArray;-><init>()V
 
     iput-object v1, p0, Lcom/mediatek/gbaservice/ril/GbaRIL;->mRequestList:Landroid/util/SparseArray;
-
-    .line 265
-    invoke-static {}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->getInstance()Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v1
-
-    iput-object v1, p0, Lcom/mediatek/gbaservice/ril/GbaRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
 
     const/4 v1, 0x0
 
@@ -2307,20 +2289,6 @@
 
     .line 709
     :goto_0
-    iget-object v1, p0, Lcom/mediatek/gbaservice/ril/GbaRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget v2, p0, Lcom/mediatek/gbaservice/ril/GbaBaseCommands;->mPhoneId:I
-
-    iget v3, p1, Lcom/mediatek/gbaservice/ril/RILRequest;->mSerial:I
-
-    iget v4, p2, Landroid/hardware/radio/RadioResponseInfo;->error:I
-
-    iget v5, p1, Lcom/mediatek/gbaservice/ril/RILRequest;->mRequest:I
-
-    move-object v6, p3
-
-    invoke-virtual/range {v1 .. v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilSolicitedResponse(IIIILjava/lang/Object;)V
-
     .line 711
     iget p2, p2, Landroid/hardware/radio/RadioResponseInfo;->type:I
 
-- 
2.53.0
